pkg/virt-api/rest/subresource.go

```diff
@@ -323,7 +314,7 @@ func (app *SubresourceAPIApp) StartVMRequestHandler(request *restful.Request, re
 				return
 			}
 		} else {
-			if vmi.Status.Phase == v1.Succeeded {
+			if vmi.IsFinal() {
```

Comment:
Is it intentionally supported now to do a start on `RunStrategyRerunOnFailure` if the VMI is in failed state, where it should be restarted anyway? It was not allowed before this commit.

Reply:
This if statement got a little bit unwieldy when incorporating the Manual RunStrategy, so I decided to do it this way.

the way it started to read was: 

```
if (runStrategy == v1.RunStrategyRerunOnFailure && vmi.Status.Phase == v1.Succeeded) || (runStrategy == v1.RunStrategyManual && vmi.IsFinal()) {
```

Which would you prefer? Including a useless condition and keeping the code a little more readable, or expanding things and hitting only the cases that matter?